cmd/openshift-install/create.go

```diff
@@ -444,12 +444,18 @@ func waitForConsole(ctx context.Context, config *rest.Config) (string, error) {
 			for _, route := range consoleRoutes.Items {
 				logrus.Debugf("Route found in openshift-console namespace: %s", route.Name)
 				if route.Name == consoleRouteName {
-					url = fmt.Sprintf("https://%s", route.Spec.Host)
+					if uri, _, err2 := routeapihelpers.IngressURI(&route, ""); err2 == nil {
+						url = uri.String()
+						logrus.Debug("OpenShift console route is admitted")
+						cancel()
+					} else {
+						err = err2
+					}
+					break
 				}
 			}
-			logrus.Debug("OpenShift console route is created")
-			cancel()
-		} else if err != nil {
+		}
+		if err != nil {
```

Comment:
any specific reason to make this change from on conditions block to 2 different ones?

Reply:
so that the existing `err` handling code can catch both the "list failed" and "route not admitted" failure modes.